### PR TITLE
Allow schemas to be loaded from a local path

### PR DESCRIPTION
**Pull request by leviathanbadger**
Breaking changes:

- Dropped support for really old versions of Node.js such as 0.10 and 0.12. (Some of the packages json-schema-validator depends on use "const" which is not supported by those versions.)

Features:

- If only one schema is loaded, use it by default if no type id is passed in to the validate method. `v.validate({latitude: 53.0, longitude: 43.0})`.
- Allow schema to be loaded from the local hard drive, not just over HTTP. This has potential security implications if the schema URL is passed unsanitized from the end user directly to json-schema-validator. Allows you to use the CLI like this: `echo '{"json": "to validate"}' | json-validate ./schema/test-schema.json`
- Lock versions of NPM packages by using package-lock.json.

CI and tests:

- Cache node_modules between runs to improve automated testing times.
- Reference a more stable online example schema for the "in the wild" networked unit tests.

Notes:

Let me know if you want me to squash these commits or change anything. I'll also understand if you don't want to accept one or more of these commits. I added these features for my own projects and use cases, and I won't be upset if you don't think some of it belongs in this repository.

### CODE REVIEW

**Comment by Magomogo** [comment]
In general I support the Idea to load schemas locally. Could you please update the Readme documentation?

**Comment by aboveyou00** [comment]
Sure thing. I've pushed a new commit. You can preview the new README.md on [my fork](https://github.com/aboveyou00/json-schema-validator/).

**Comment by aboveyou00** [comment]
Note: I just fixed up the two feature commits to clean up the code. Reasons:

1. Remove a `console.log` that I was using for debugging purposes, but that shouldn't be in production code.
2. Remove an unnecessary typecheck. (`data = data.toString(); if (typeof data === 'string') ...`)

The new commits are not substantially different.

**Review by Magomogo — CHANGES_REQUESTED** [review]

File: src/validator.js [hunk 1]
```diff
@@ -74,39 +75,87 @@
...
+                callback(null, data);
+            });
+        }
+    }
 
-                    callback(null, body);
-                });
-            },
+    Validator.simple = function (schemaUris, callback) {
+        if (schemaUris === null) {
```
[the remark this answers is not in the document]
Comment (aboveyou00) [reply src/validator.js]: I accidentally passed null while writing one of the unit tests and the error message was less than helpful. (Specifically, there was none. It just pretended that it worked and allowed the validator to fail later when it had no schema to use. It turns out `async.map` treats null as an empty array. Undocumented, of course.)

I even added a small unit test to verify this check. Do you want me to remove it?

Alternatively, it occurs to me that this error would be more helpful in the `fetchSchemas` method. I could move it there instead...

[the remark this answers is not in the document]
Comment (aboveyou00) [reply src/validator.js]: Note: this change request is on an old commit.

[the remark this answers is not in the document]
Comment (Magomogo) [reply src/validator.js]: Ok, I see.

(I'm still not used to this github code review interface, I did this comment yesterday, but haven't submitted it)

**Review by Magomogo — APPROVED** [review]

**Comment by Magomogo** [comment]
Thanks, Brandon! Do you want me to release a version or you have something to add?

**Comment by aboveyou00** [comment]
I've stopped working on it for my own project, so unless you have something specific you'd like to request then creating a release is fine.

**Comment by Magomogo** [comment]
@aboveyou00 enjoy https://github.com/Magomogo/json-schema-validator/releases/tag/v1.2.0
:)

**Pull request merged**